### ISSUE TITLE
dotnet_style_prefer_inferred_anonymous_type_member_names = false does not produce warning

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#29945]
I have following test code:

```csharp
var objectsWithMeta = DataContext.Objects.Select(o => new { Object = o, Metadata = o.Metadata }).ToList();
var objectsWithMeta2 = DataContext.Objects.Select(o => new { Object = o, o.Metadata }).ToList();
```

With `dotnet_style_prefer_inferred_anonymous_type_member_names = true:warning` in my `.editorconfig` file,
I get following warning for the first line: &quot;Warning	IDE0037	Member name can be simplified&quot;
Just as expected.

Now, if I change to `.editorconfig` file to `dotnet_style_prefer_inferred_anonymous_type_member_names = false:warning`
I expect some warning for the second code line but there is none.

Is this the designed behavior? How can I get some warning if members are not named explicitly?

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/336226/dotnet-style-prefer-inferred-anonymous-type-member.html
VSTS ticketId: 684966_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by CyrusNajmabadi** [comment]
This is expected behavior.  The preference is simply about if you should get a warning or not when too verbose.  In a sense the feature could be thought of as "require inferred member name when possible". and you are saying yes/no in terms of if the inferred-name is required.  When you say "no, it is not required", that is not equivalent to saying "no, it is not allowed.".

**Comment by gojanpaolo** [comment]
Hello, how can we enforce explicit member names for anonymous types? Thanks!

**Comment by CyrusNajmabadi** [comment]
@gojanpaolo You can do this by writing your own analyzer for this case.  Thanks! :)

**Comment by gojanpaolo** [comment]
@CyrusNajmabadi got it! thanks!

**Comment by maknapp** [comment]
Should setting `dotnet_prefer_inferred_anonymous_type_member_names` to `false:silent` hide the suggestion? No matter what I set in the .editorconfig file, visual studio always displays the suggestion "IDE0037 Member name can be simplified" in the Error List window.

I set `dotnet_prefer_inferred_tuple_names` to `false:silent` and the suggestion for that is removed, so the editorconfig file is working. Is there some other setting the overrides it?
